File: app/src/main/java/com/esorokin/lantector/di/module/RequeryModule.java
package com.esorokin.lantector.di.module;

import android.content.Context;

import com.esorokin.lantector.model.data.Models;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import io.requery.Persistable;
import io.requery.android.sqlite.DatabaseSource;
import io.requery.sql.EntityDataStore;

@Module
public class RequeryModule {
	public static final String DB_NAME = "lantector.db";
	public static final int DB_VERSION = 1;

	@Provides
	@Singleton
	DatabaseSource provideSource(Context context) {
		return new DatabaseSource(context, Models.DEFAULT, DB_NAME, DB_VERSION);
	}

	@Provides
	@Singleton
	EntityDataStore<Persistable> provideStore(DatabaseSource databaseSource) {
		return new EntityDataStore<>(databaseSource.getConfiguration());
	}
}

File: app/src/main/java/com/esorokin/lantector/di/module/RequeryModule.kt
package com.esorokin.lantector.di.module

import android.content.Context
import com.esorokin.lantector.model.data.Models
import dagger.Module
import dagger.Provides
import io.requery.Persistable
import io.requery.android.sqlite.DatabaseSource
import io.requery.reactivex.KotlinReactiveEntityStore
import io.requery.sql.KotlinEntityDataStore
import javax.inject.Singleton

@Module
class RequeryModule {
    companion object {
        const val DB_NAME = "lantector.db"
        const val DB_VERSION = 1
    }

    @Provides
    @Singleton
    fun provideSource(context: Context): DatabaseSource = DatabaseSource(context, Models.DEFAULT, DB_NAME, DB_VERSION)

    @Provides
    @Singleton
    fun provideStore(databaseSource: DatabaseSource): KotlinReactiveEntityStore<Persistable> = KotlinReactiveEntityStore(KotlinEntityDataStore(databaseSource.configuration))
}
